### PR TITLE
Fix check-in, check-out bugs

### PR DESCRIPTION
**Pull request by liayoo**
- Add sender_proof to check-in requests to make sure that the requester actually owns the 'sender' ethereum key
  - sender_proof is a message containing the ref, amount, sender, timestamp, and nonce that's signed by the sender eth key
- Check that check-in, check-out ids don't exist in the history
  - Note that this alone cannot enforce unique keys though, since history is garbage collected. I'll update the bridge server to close a request with status = false if there's already a record in history with the same id
- Change check-in, check-out request status type to boolean

### CODE REVIEW

File: db/functions.js [hunk 1]
```diff
@@ -964,6 +966,24 @@ class Functions {
...
+  validateCheckinSenderProof(ref, amount, sender, senderProof, chainId, tx) {
+    try {
+      const body = {
+        ref,
+        amount,
+        sender,
+        timestamp: _.get(tx, 'tx_body.timestamp'),
+        nonce: _.get(tx, 'tx_body.nonce'),
+      };
+      if (!ainUtil.ecVerifySig(stringify(body), senderProof, sender, chainId) === true) {
```
Comment (platfowner) [inline db/functions.js:978]: question: is this correct that 'sender' is a ethereum address but 'sendProof' is generated using ain-util (not ethereum signing library)?

Comment (liayoo) [reply db/functions.js]: You're right! I've updated to use web3-eth-accounts's sign & recover functions. Thanks!

File: db/functions.js [hunk 2]
```diff
@@ -991,7 +1012,7 @@ class Functions {
     const tokenId = context.params.token_id;
     // NOTE(liayoo): `sender` is the address on `networkName` that will send `tokenId` tokens to the pool.
     // For example, with the Eth token bridge, it will be an Ethereum address that will send ETH to the pool.
-    const { amount, sender } = value;
+    const { amount, sender, sender_proof: senderProof } = value;
```
Comment (platfowner) [inline db/functions.js:1015]: could you add a note for the 'senderProof' as well?

File: tools/checkin/sendOpenCheckinTx.js [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
+  const ref = `/checkin/requests/ETH/3/0xB16c0C80a81f73204d454426fC413CAe455525A7/${fromAddr}/${checkinId}`;
+  const timestamp = Date.now();
+  const body = {
+    ref,
+    amount: tokenAmount,
+    sender: config.senderAddr,
+    timestamp,
+    nonce: -1,
+  };
+  const senderProof = ainUtil.ecSignMessage(stringify(body), Buffer.from(config.senderPrivateKey, 'hex'), 3);
```
Comment (platfowner) [inline tools/checkin/sendOpenCheckinTx.js:19]: need to be updated as well?

Comment (platfowner) [reply tools/checkin/sendOpenCheckinTx.js:19]: and could you add a note for 'senderProof' here as well?

File: db/functions.js (hunk 2 above)
Comment (liayoo) [reply db/functions.js:1017]: done!

File: tools/checkin/sendOpenCheckinTx.js (hunk 3 above)
Comment (liayoo) [reply tools/checkin/sendOpenCheckinTx.js]: right, thanks! done!

**Review by platfowner — APPROVED** [review]
LGTM
Thanks for handling the issues.

**Pull request merged**